src/handlers/authentication.ts

```diff
@@ -36,7 +36,7 @@ export async function savePIN(pin: any) {
   }
 }
 
-export async function authenticateWithPIN() {
+export async function authenticateWithPIN(safeIfNeed: boolean = false) {
```

Comment:
`safeIfNeed`? what do you mean?

Reply:
if I pass this prop, the pin will be saved if we haven't the previous one
I will replace it with `shouldSavePin`